netcore/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -46,11 +44,7 @@ static class CustomAttribute
 		static bool IsUserCattrProvider (object obj)
 		{
 			Type type = obj as Type;
-#if !FULL_AOT_RUNTIME
 			if ((type is RuntimeType) || (RuntimeFeature.IsDynamicCodeSupported && type is TypeBuilder))
-#else
```

Comment:
A linker friendly way is to add an internal virtual IsTypeBuilder method in the Type class which returns false, and overwrite it in TypeBuilder.

Reply:
Thanks, added.